rollout/bluegreen.go

```diff
@@ -223,6 +227,14 @@ func (c *RolloutController) syncRolloutStatusBlueGreen(previewSvc *corev1.Servic
 	oldRSs := roCtx.OlderRSs()
 	allRSs := roCtx.AllRSs()
 	newStatus := c.calculateBaseStatus(roCtx)
+
+	if replicasetutil.CheckPodSpecChange(r, newRS) {
+		roCtx.PauseContext().ClearPauseConditions()
+		roCtx.PauseContext().RemoveAbort()
+		newStatus = c.calculateRolloutConditions(roCtx, newStatus)
+		return c.persistRolloutStatus(roCtx, &newStatus)
```

Comment:
Actually, I don't think we should return early here since we definitely want to set the ActiveSelector, HPAReplicas, and HPAReplicas values. I think the previewService is okay to set since the next reconciliation will change it when the newRS is created (because the rollout is now unpaused). Lastly, the calculateScaleUpPreviewCheckPoint function will return false since it checks for a Pod Spec change which is what we would want for ScaleUpPreviewCheckPoint.

Reply:
Has the expected selector state of the previewService changed between 0.5.0 and 0.6.0?

- In 0.5.0, the previewService will no longer select green (newRS) nor blue once unpaused.
- In 0.6.0, the previewService will continue to select green once promoted.

Along the same lines, what is the expected behavior of the previewService on rollback?

- Is it true in 0.5.0 that the previewService will no longer select green (newRS) nor blue on rollback?
- I assume in >0.6.0, the previewService will continue to select green on rollback???

Reply:
@carlosdoordash, your first two bullets points describe the behaviors for the two versions correctly. 

Before I answer your two questions, I am assuming that by green, you mean the new version that needs to be rolled back, and blue is the healthy version. Please let me know if that assumption is incorrect. 

```
Is it true in 0.5.0 that the previewService will no longer select green (newRS) nor blue on rollback?
```

Yes, this was the behavior for v0.5 as the blue version (now becomes the newRS) is automatically promoted to active service, and the preview service's selector points at nothing.

```
I assume in >0.6.0, the previewService will continue to select green on rollback???
```

For rollback in v0.6, the blue version (now becomes the newRS) is automatically promoted to the active service and the preview service points at the ReplicaSet that matches the current Rollout Spec (which is blue in this case).

Please let me know if I should clarify more!